pkg/registry/interface.go

```diff
@@ -12,6 +12,7 @@ type Load interface {
 	AddPackageChannels(manifest PackageManifest) error
 	AddBundlePackageChannels(manifest PackageManifest, bundle *Bundle) error
 	RemovePackage(packageName string) error
+	DeprecateBundle(csvName, version, path string, tailBundles []string) error
```

Comment:
I can see the use case of wanting to query the list deprecated bundles for development reason or pipeline work.
For dependency resolution, should we modify the `ListBundles` query to exclude the deprecated bundles given those shouldn't considered for dependency resolution right? @ecordell @benluddy

Reply:
I opened https://github.com/operator-framework/operator-lifecycle-manager/pull/1673 to filter deprecated bundles in the resolver

Reply:
I saw that PR but I wonder if we need to do that on resolver side when we can do it on the `ListBundles` query itself.

Reply:
@ecordell made a point that it's easier to undo if it's on the resolver side